### DISCUSSION TITLE
Process for refactorings for 3.x and patch releases for 2.5

### DISCUSSION THREAD

**Discussion by kaikreuzer** [openhab/openhab-distro#1506] [openHAB Maintainers Discussions]
Hi @openhab/maintainers,

Now that the 2.5 release is done and our master branches have become 3.0.0-SNAPSHOT, we will need to discuss how to best proceed.
One technical (and legal) necessity is to refactor our openhab-core code from `org.eclipse.smarthome` to `org.openhab.core` namespace. Doing so will mean that all add-ons will need to be adapted at the same time and will thus immediately become incompatible with a 2.5 runtime - including the 130+ open PRs that still exist on openhab2-addons.

As I have mentioned in the release blog post, I don't think we should do a 2.6 release, but I do believe that we will have to ship 2.5.x versions of the add-ons, since it will take too long until a 3.0 release becomes available. We could create a 2.5.x branch in openhab2-addons (and maybe a 1.14.x one in openhab1-addons) and merge any PR that is not breaking any backward compatibility against this branch. From there, we could still build addons.kar and publish online-repos with those versions. It would imho also be ok to merge new bindings there (for which we still have plenty of PRs).

If we go this route, the question is: When and how do we refactor the add-ons to the new namespace on the master branch? Can we come up with a clever script that does this automatically for us and we can do bulk syncs from 2.5.x to master from time to time? Or shall we first completely remove the bindings from master, work on the 3.0.0 build (with new namespace, new UI, etc.) without any bindings and then do the migration as a single operation (and from then on require PRs against master and potentially separate backports against 2.5.x)?

What's your opinion on the best process here?

**Comment by J-N-K** [comment]
That's a difficult question. I think scripting the namespace-change should be possible.

But we talked about JDK11 and also about moving some bundles (like mqtt) from core to addons. Both will break compatibility, i.e. a MQTT3 binding-bundle will differ from MQTT2.5.x binding bundle not only by different namespaces (not sure about other bindings/core-bundles). 

Going with option 2 is not ideal, too. We can't test 3.0 without bindings and that will probably leave bugs unnoticed before we move addons.

**Comment by kaikreuzer** [comment]
The "classic" way would be to just keep going and do manual backport of certain commits to the 2.5.x branch - or a manual forward-port of existing PRs that we merge against 2.5.x. This would probably be most flexible, but also most effort. And suitable "namespace switch" scripts would be very handy for such an exercise.

**Comment by Hilbrand** [comment]
Do we know or has someone an overview of all what needs to renamed? What the impact is? Like for jsondb? rules? etc? I haven't seen an issue :wink: How about namespace of compatibility mode? Does that come into conflict when we rename to openhab? If we know that we identify how much effort it will be. Only renaming can be very likely completely scripted (like I did for for the bnd migration, were I had a script to migrate completely automatically and also a script to completely fix it when there was a merge conflict on a migration pr).

I personally would prefer to migrate as soon as possible. Otherwise we keep maintaining different realities. I expect the rename to be relatively simple. I would even suggest to just release this as 3.0 so we can move on. I realize this would be a bit strange as Kai has up-ed the expectations for 3.0 in the blogpost by promising a new UI (which I'm not sure how this will unfold given the current development of the ui (see my [comment on the ui repo](https://github.com/openhab/openhab-webui/issues/24#issuecomment-566296208)) So an early release of 3 would bring nothing new for the user while suggesting a big change because of the new number. Another option could be to release it as 2.99 or something?

Another option could be to do the rename just before a final 3.0 release when all other changes are made. If it's a scripted process that can be done in short time this would be an option. However it also mean we need some kind of roadmap with what would actually be in 3.0 other than the rename. And except for wishes I don't know of any large new features that can be guaranteed given it's all voluntarism...

With regards to the open pull request. I suspect there will keep being pull request so that will always be an issue I think. However we could merge or handle the enhancement/bug fixes as those are probably the least convenient to change after a rename. New bindings might be easily script migrated were either we could offer help with a script or provide the script. Also we've been tagging a number of older pr's in recent months with stale, so I suspect we can clean up on those before and see for the other new binding pr's if we can merge as much as possible first.

**Comment by kaikreuzer** [comment]
> Do we know or has someone an overview of all what needs to renamed?

Well, as I said: Everything that currently uses `org.eclipse.smarthome`. And this should be the first activity we now do on the new master - which will break the build as long as we only do it for openhab-core as a start.

> How about namespace of compatibility mode?

This will indeed collide and another good reason to remove it at the very same time.

> I personally would prefer to migrate as soon as possible. Otherwise we keep maintaining different realities.

As mentioned: We will have to maintain both branches for a while, no way around that. So the question is only, what process is better: To maintain a single one and bulk port stuff or to decide on every single PR, where to merge it and whether to back/forward port it.

> I expect the rename to be relatively simple.

Is anyone scripting-fluent enough to come up with a script for the namespace change? Which could also be applied on a single binding, e.g. after having merged it in the 2.5.x branch and needing to forward port it to master? Maybe @wborn? 🤞

**Comment by kaikreuzer** [comment]
To come with a more concrete question/decision:
Would you all be ok with:
- creating a 2.5.x branch on openhab2-addons and openhab-webui and actively merge existing backward-compatible PRs against this new branch, while manually porting them to master
- provide a 2.5.1 patch release of ONLY the addons.kar and the online-repo (no new runtime/distro being built).
- asap refactor openhab-core, openhab2-addons & openhab-webui to the new namespace and remove the compat layer
- finish https://github.com/openhab/openhab2-addons/pull/5275
- replace dashboard, Paper UI, Classic UI, HABmin in openhab-webui by the new UI

**Comment by BClark09** [comment]
In terms of packaging I think it would make sense to stop linuxpkg builds until the 2.5.x branch is setup in the way above. A new package needs to be setup for openHAB 3.x so that needs to be thought of carefully too. 

Should I discuss options for this in a new @openhab/maintainers post?

**Comment by kaikreuzer** [comment]
I have deactivated the linuxpkg build for the time being.
For options wrt linux packaging, I'd rather suggest the discussion to take place in https://github.com/orgs/openhab/teams/linuxpkg-maintainers. Feel free to invite other people to it that you feel can help on the discussion, though.

**Comment by Hilbrand** [comment]
If we want to keep both versions for now we could also do the following with the openhab2-addons:
1. Create a new branch 3.0.x
2. Fix the master branch to use 2.5.0 openhab-core
3. Script the migration of smarthome to openhab on the 3.0.x branch and git reset that branch to the master and rerun the script (for example each night like a snapshot build)
4. When we make the actual switch to 3.0 and not maintain 2.5, the 3.0.x branch can become a pr to the master and with that the migration will be completed.

This would make it easy to maintain, but will break when we would make breaking changes in core that would require bindings to be updated (something that would be a problem also in any scenario). One could argue that with the master isn't 3.0 and 2.5.x is still leading. This would also mean new bindings would still always be developed against 2.5 and no pr's should be made against the 3.0.x branch as it's reset everytime.

Here is a script for core: https://gist.github.com/Hilbrand/94c3a1049a494d763223e2d33f4e8e73
Maven build will break on the compatx and itest for persistence. I haven't tested if it actual runs. (and it will not be compatible with any existing jsondb configuration obviously)

Regarding removing compat layer. That is a big change and will probably not completely be a surprise, but still be somewhat unexpected. if we do that we should startup a migration program (Looks like I need to finish my tutorials :smile: ) Another option could be to move compat to a package including `legacy`.

While we're at it. Can we also replace Nashorn with Graal for the rule engine?

**Comment by cdjackson** [comment]
Throwing my few cents worth into the pot :)

I don’t mind too much what approach we take to maintain the 2.5 branch so I’ll not comment on that. And I think we’re probably getting off topic here when we talk about the compatability layer - however, I’ll take it a little more off topic (sorry)… 

I think if we’re on OH3, then maintaining compatibility with OH1 is getting a bit long, and we should consider having some larger discussions about resolving broader issues with the architecture. I’m not sure if there is another discussion forum about the OH3 direction, but IMHO we should look to have a broader discussion before jumping in and building something. I’m sure people would love to maintain OH1 compatibility, but we should take care in that by trying to maintain such compatibility we will limit what can be done in OH3.

If there a place where we should discuss this direction, or has this already been done?

**Comment by bgilmer77** [comment]
@cdjackson, thanks for the excellent observations about getting a discussion going regarding whether OH3 is a breaking change from OH1.  This seems to be a governance question of high importance.  Do I remember correctly that a council was created for this purpose?  Perhaps I am thinking of an architecture council.  In this case, I would strongly argue for inclusion of users in the discussion, specifically users who are not developers.  (Yes, I am volunteering if you want me...).  So to your question, where should this be discussed?

As for maintaining backward compatibility with OH1, my feeling is that OH1 adopters are significantly different from OH2 adopters.  Specifically, it took some pretty hard work to get OH1 working, so I suspect most of those people were/are technical, compared to OH2 users, who may still be pretty technical, but not as hardcore as OH1 folks.  If I am right, then breaking compatibility may result in inconvenience, but the deployed base is (probably?) relatively small, and theoretically, those who have it have the technical chops to deal with the change.  But, and this is a very big but, if this were my business, I would definitely want to verify this hypothesis by some means, perhaps surveys?  other ideas?  but anyway, it seems worth the effort to not make a mistake.

Also breaking changes and the perceptions they create can be difficult (impossible?) to correct after the fact.  So I would strongly suggest that there needs to be a lot of thought about messaging, tutorials, documentation and support for getting from OH1 to OH2/3, etc.  It is worth getting this right.

My two cents...

**Comment by bgilmer77** [comment]
Oh yeah - one other thing.  What are the end-user driven use cases or requirements behind needing to make a breaking change?  It is worth putting those down somewhere.  If for no other reason, it helps us explain why we "broke OH1" (yes - some people might say that).  Perhaps it is clear to all now, but trust me, two years from now, if something comes up to bite us, we will appreciate knowing why we did this.

**Comment by kaikreuzer** [comment]
Guys, this thread is NOT about directions for openHAB3. Our master now IS openHAB 3 development and we should move towards getting snapshot builds working for it asap.
There are two main issues that I want to solve with this thread here:
1. We HAVE TO change the namespace asap. This will require quite some changes and especially break existing openhab2-addons PRs.
2. Many PRs on openhab2-addons have already been merged on master (i.e. 3.0.0-SNAPSHOT). Many if not all of them will have to be made available in a 2.5.1 release to existing 2.x users.

This is what [my suggestion](https://github.com/orgs/openhab/teams/maintainers/discussions/2/comments/5) wants to address. Whether or not to restore 1.x compatibility and how can be decided separately (and yes, there has been plenty of discussions already and the final decision will be taken by the AC).

**Comment by kaikreuzer** [comment]
@Hilbrand Thanks for the script, that looks like a good starting point!

**Comment by cdjackson** [comment]
Hi Kai,
I agree - and I apologise (again) for getting off topic, even if the discussion was heading toward the removal of the comparability layer.

However, I'd still like to have a discussion about what OH3 is trying to achieve - I'm not aware of any previous discussions (other than a few ad-hoc points) and I think before starting out on a development, we really should understand WHAT we are trying to develop.

Or are you basically just saying that OH3 is OH2 with a change of the ESH namespace? Or is this not up for discussion?

So, you point out -:

> this thread is NOT about directions for openHAB3

But my earlier question still remains -:

> Is there a place where we should discuss this direction, or has this already been done?

Chris

**Comment by kaikreuzer** [comment]
> Or are you basically just saying that OH3 is OH2 with a change of the ESH namespace? 

As a very first step to get snapshot builds working again: Yes, that's about it!

> what OH3 is trying to achieve - I'm not aware of any previous discussions 

There was already a lot of discussion, see https://community.openhab.org/t/directions-for-openhab-3/67372 for the general direction and https://community.openhab.org/t/removal-of-the-oh-1-x-compatibility-layer/67553 on the future of the compat layer.

**Comment by cdjackson** [comment]
I really think we should have a consolidated set of "requirements" and some sort of agreed direction. I'm not sure that opening up such a discussion to the whole forum would be very fruitful, but I do think that for OH to survive and flourish we should look to fix some of the fundamental issues we have rather than retail all the same basic concepts as we had 10 years ago. I'm happy to participate in such a discussion if desired - I'm supporting two of the biggest bindings, and also support 4 commercial systems based on OH/ESH (and a lot more that don't use OH) so have a reasonable amount of visibility of issues people are having.

I'll not clog up this thread any more - again sorry for doing so, but I would strongly suggest that we do have some level of planning/direction/requirements/etc so that we make the system better and not just another version of OH1 with a new UI.

**Comment by wborn** [comment]
I would very much prefer to have only one active (master) branch where we work on 3.0.0.

For the bigger part of this year I've been working with two long running active branches on another project and it was not good for my emotions after several months. At least I got compensated for that with €. :-)

I can only fear what the result will be with multiple active branches on a project like openHAB where there are lots of contributions and we currently don't even have enough manpower for handling the PRs of a single active branch.

So I would like to continue as we have always done. Then if the need arises for some in-between release/add-on pack, we backport the 3.0.0 add-ons from the master to 2.5.x using some helper script.

This way the community/issues/PRs won't be split in two, which will save a lot of confusion, work and frustration.

**Comment by kaikreuzer** [comment]
@wborn While I very much understand that wish, I am afraid that we won't get around providing a 2.5.1 for add-ons at some time. I would not expect to have an openHAB 3 release before end of 2020 and we already see many bug reports in bindings of 2.5, for which fixes are created right now. Providing a patch for the existing bindings should take some pressure from us wrt the 3.0 release. But yes, we can certainly say that we only pick selected major/critical issues to backport to the 2.5.x branch.
We nonetheless also have many bindings that are more or less done, but which didn't make it into the release (and the respective contributors are unhappy about that). So if we ship a 2.5.1 anyhow, we could also think about including those new bindings instead of only adding them to 3.0. But yes, this can be discussed and you are probably right that it could lead to two very active parallel streams, which I am not fond of either.

**Comment by Hilbrand** [comment]
Do we expect changes to the interface used by bindings with core classes? Otherwise we could just keep a 2.5.1 branch and auto migrate it on regularly to a 3.0.0 branch where the only differences is the namespace. That way we only have to maintain one branch.

**Comment by kaikreuzer** [comment]
That's what I meant with the "we can do bulk syncs from 2.5.x to master from time to time" option above.
I don't necessarily expect breaking changes within the next weeks or at least we can probably easily avoid them.

**Comment by kaikreuzer** [comment]
Ok, as more and more PRs get merged on openhab2-addons, I will now go for

> creating a 2.5.x branch on openhab2-addons and openhab-webui 

and @cdjackson, I assume we should do the same for zigbee and zwave or don't you expect any fixes to be made available on 2.5.x anymore? At least zwave db updates will probably make sense, I guess.

**Comment by cdjackson** [comment]
Hi Kai, 
Yes, I think for the foreseeable future I will likely only develop for the 2.5 branch of these bindings until there is a clear direction for 3.0 as I'm already getting grief from people and I will clearly need to provide support and enhancements on 2.5. Since the only change seems to be a namespace change, hopefully the script can resolve this.
Chris

**Comment by kaikreuzer** [comment]
Ok - I have created a 2.5.x branch for openhab2-addons, openhab-webui, openhab-bundles, org.openhab.binding.zwave, org.openhab.binding.zigbee and openhab-distro now and will try to get a snapshot build working under https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/.
I have also created a 2.5.1 milestone in Github for all of these projects. So PLEASE: Any issue that you backport from master to the 2.5.x branch, please assign it to the 2.5.1 milestone.

**Comment by cdjackson** [comment]
Will CI build a nightly version of 2.5.x? If we migrate the master branch to OH3, it would still be good to support a 2.5.x-SNAPSHOT if we can?

**Comment by kaikreuzer** [comment]
@cdjackson Just read my comment that is directly above yours and you will find the answer: https://github.com/orgs/openhab/teams/maintainers/discussions/2/comments/23

**Comment by cdjackson** [comment]
Sorry - I didn't refresh the page and this doesn't seem to update when new posts are created.

Thanks.

**Comment by kaikreuzer** [comment]
All, we have now green 2.5.1 builds for the mentioned repos at https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/, which means that we have KAR files for all add-ons available and the [snapshot online-repo](https://openhab.jfrog.io/openhab/online-repo-snapshot/2.5/) being populated with 2.5.1-SNAPSHOTs of the add-ons. I'd like to setup a sandbox-release build as well in the coming days, so that we can easily do a 2.5.1 release and switch to a 2.5.2-SNAPSHOT. I would suggest to consider the 2.5.x releases being done regularly, possibly also on a monthly bases as the milestones, wdyt?

Now that https://github.com/openhab/openhab-core/pull/1284 is merged, I think we are ready to go for the namespace refactoring in openhab-core on master. Any volunteer for this task?

**Comment by wborn** [comment]
As I have commented in [#6636](https://github.com/openhab/openhab-addons/pull/6636) it is still unclear to me:

* what versions we will be working on
* what scope they have (bug fixes/features/major breaking)
* how they will be released (online repo/KAR/full release)
* when they are intended to be released
* how we will work on them (branching strategy) per repo

It would already help to clarify when this in-between release is scheduled and what should be part of it.

We now have 2.5.x branches but as you can read in the comments above there has never been an agreement or vote on this.

**Comment by kaikreuzer** [comment]
I think most of those questions are answered above.
- Versions: 3.0.0 and 2.5.x
- 2.5.x for non backward-compat breaking changes
- Releases as KAR and online-repo
- 3.0.0 Milestones and 2.5.x patches potentially on a monthly basis or as needed (can imho be decided, when necessary)
- branching strategy to be discussed by the repo maintainers. Chris decided for a different strategy for zwave & zigbee than what would make sense for openhab-addons, and that's fine.

Right, I didn't ask for a formal voting, but rather tried to be pragmatic and do suggestions to which I didn't see a veto - only you mentioned to prefer to work only on master, but I answered that one and did not feel anybody questioning the need for 2.5 updates.

If you feel we need a more formal process in future, please suggest how to structure and organize that. My personal feeling was that it had to be done ad-hoc as work continued on the repos already with PRs being merged etc. and I just wanted to get this aligned asap.

**Comment by cdjackson** [comment]
Personally, I'm fine with the approach - thanks for setting everything up Kai.

Maybe I missed it somewhere, but was a script finally produced to do the namespace changes or is that still an ongoing discussion?

**Comment by davidgraeff** [comment]
Personally I will not backport any reviewed & merged PRs. That's way too much effort and developers and maintainers are probably on master only.

Also the current issue board of Github is usually for the master branch. We would need to add a label "2.5 branch" or similar to label issues that are actually resolved in master but not yet backported or are only valid for 2.5 because the respective binding had been refactored in 3.0 etc.

**Comment by wborn** [comment]
I second @davidgraeff with this. This approach is unsustainable for openhab2-addons. It will more than double the workload of our maintainers. They will have even less joy in their already unthankful work. This may also make them abstain from processing PRs or stop being a maintainer. As a result PRs will not be merged and users are not served well by this.

**Comment by digitaldan** [comment]
Would it possible to put the burden of back porting on the PR owner and have a semi automated way of back porting changes from master to 2.5?  For example, if I submit a PR which i would like in both master and 2.5.x i would submit a PR as normal against master, but tag it with "backport-2.5".  Once the PR has been reviewed and merged, something like https://github.com/marketplace/actions/backport-bot creates a new PR to 2.5.x (based on the PR having the tag) .  If the PR passes Travis build tests then it could be merged without further review (maybe?)

**Comment by J-N-K** [comment]
Changing the base for the PR to merge to 2.5.x won't work. I just tried that and eded up with 286 changed files for the PR (which was a three-line-change).

**Comment by kaikreuzer** [comment]
This seems to be a openhab2-addons repo only discussion, so I'd suggest to move it there: https://github.com/orgs/openhab/teams/add-ons-maintainers/discussions/6

**Comment by kaikreuzer** [comment]
Back to the main stream here, just to bring it back:

> Now that openhab/openhab-core#1284 is merged, I think we are ready to go for the namespace refactoring in openhab-core on master. Any volunteer for this task?

I can have a try with @Hilbrand's script, but if anybody else wants to go for it, it is very welcome as well.

**Comment by kaikreuzer** [comment]
I have created an issue for tracking: https://github.com/openhab/openhab-core/issues/1290

**Comment by pfink** [comment]
@kaikreuzer: I could do the work on sandbox-release. I already splitted the job and just started a test run for the 2.5.x branch.

2.5.x -> https://ci.openhab.org/view/Sandbox/job/sandbox-openhab2-release/
3.x -> https://ci.openhab.org/view/Sandbox/job/sandbox-openhab3-release/

For now I disabled the timer of the 3.x job.

**Comment by kaikreuzer** [comment]
Excellent, thanks @pfink!

**Comment by cweitkamp** [comment]
> I can have a try with @Hilbrand's script, but if anybody else wants to go for it, it is very welcome as well.

I gave it a first try: https://github.com/openhab/openhab-core/pull/1294

I had a green build locally from my console (Linux) with one failing itest (`org.openhab.core.model.script.tests`). I will work on that later. Testing the build on other platform would be appreciated. Lets see what Travis and Jenkins will say.

Next step for me will be to get a running build in Eclipse IDE.

**Comment by wborn** [comment]
Is there also a documentation/website branching strategy? I don't see any branches for these projects. 

It might be confusing if the docs are updated for Java 11 changes (https://github.com/openhab/openhab-core/issues/1287) that are not applicable to 2.5.x. Does the "latest" docs on the website show the documentation of the 2.5.x or master branches?

**Comment by kaikreuzer** [comment]
I already asked @ghys to check that no changes on master are published in the docs - imho, the main version of the docs should stay on 2.5.x until 3.0 is released - otherwise it will cause too much confusion.

A bit more tricky might be the developer documentation section, but probably also for this it makes sense to mainly show 2.5.x for now, so that would be in line.

**Comment by kaikreuzer** [comment]
I have heavily worked on figuring out how we can provide 2.5.1 add-ons to existing 2.5.0 distro users and had to learn that it does not work as smoothly as I had thought. My assumption was that we can simply provide the newer add-ons under the existing 2.5 online-repo url that is configured in the distro and that it would automatically pick them up.
Unfortuntely, the distro contains the exact version 2.5.0 of the add-ons to use, so that this does not work.
Removing the 2.5.0 add-ons feature in the distro and adding a new 2.5.1 one doesn't work either as the distro does not allow to uninstall the old feature, while any contained feature is still installed - and as the dashboard itself is a part fo the add-ons feature, there isn't any decent way to uninstall them.

Doing many attempts, I came up with a solution, though, for which I have created https://github.com/openhab/openhab-distro/pull/1041. This solution means that we will have to again ship a 2.5.1 distro as an update. But for any further 2.5.x, all we need to do is to publish add-ons to Bintray and they will be automatically used by the existing installations.
While being at it and taking the chance to roll out a further distro, I have also fixed https://github.com/openhab/openhab-distro/issues/961 and https://github.com/openhab/openhab-distro/issues/1031 on the way. 
I hope you are all fine with that approach, it is the best I was able to come up with 🤔.

**Comment by kaikreuzer** [comment]
Ok, I didn't see any veto and the PRs have been merged - as a result, we have a nicely working snapshot distro at https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/job/openHAB2.5.x-Distribution/. I have also adapted the linux packaging and it should be available as well.
I'd suggest to do a 2.5.1 release on Jan 12, so that we have the major flaws of the distro (like https://github.com/openhab/openhab-distro/issues/1031) fixed and no beginners stumbling over them anymore.
So please consider this date and make sure to only merge stuff into 2.5.1 which is backward compatible and not too risky.

**Comment by wborn** [comment]
Will there continue to be 2.5.x-SNAPSHOT .tar.gz/.zip files or only KAR files after the 2.5.1 release?

**Comment by kaikreuzer** [comment]
The 2.5.1 distro should be enough (unless we find other major issues we need to fix in the distro itself). So my plan would be to only build KARs and publish the add-ons on Maven as they will then become available in the 2.5.1 distro.

**Comment by BClark09** [comment]
Thanks @kaikreuzer 

> plan would be to only build KARs

When that happens, I'll change Linuxpkg's Jenkins build to only package and release 2.5.>1 versions of `openhab2-addons` for apt and yum.

**Comment by cweitkamp** [comment]
> Next step for me will be to get a running build in Eclipse IDE.

With https://github.com/openhab/openhab-distro/pull/1030 in place and the [latest fix](https://github.com/openhab/openhab-core/pull/1315) from @wborn I have a running headless OH3 distribution in my IDE using Java 8.

How should we continue?

I want to test the switch to Java 11 and I am planning to move the `ESH-INF/` folders to `OH-INF/ ` folders.

I already ported some add-ons (e.g. DarkSky and Google TTS) locally and started migration of Pushover OH1 action to OH3 binding (see https://github.com/openhab/openhab-addons/pull/6798). All of them seem to work fin when running from my IDE.

**Comment by kaikreuzer** [comment]
> I want to test the switch to Java 11 and I am planning to move the ESH-INF/ folders to OH-INF/ folders.

Sounds absolutely fine to me as next steps!

**Comment by kaikreuzer** [comment]
Heads-up: If there's no veto, I will do the 2.5.1 build tomorrow morning.

**Comment by kaikreuzer** [comment]
@J-N-K and others: Please refrain from merging PRs until the build is successfully done - https://ci.openhab.org/job/openhab-release/72/

**Comment by J-N-K** [comment]
Sorry. Didn‘t see that. Will stop now.

**Comment by J-N-K** [comment]
It seems it seems me and Jenkins messed up the master branch. How do we fix that?

**Comment by kaikreuzer** [comment]
You mean the 2.5.x branch with https://github.com/openhab/openhab-addons/commit/e2b76330982f2658e968a58522b4aede248fd487, right? I will remove this commit and do a force push. @pfink, looks as if we need to do slight changes to the jenkinsfile...

**Comment by kaikreuzer** [comment]
Ah, no - looking closer at it, Jenkins did everything correctly - it made sure that the PRs that you had merged after the release build had already started are correctly put on top of the version change to 2.5.2.
So do you think there's anything wrong that needs to be corrected?

**Comment by J-N-K** [comment]
The single commits are lost due to the merge commit. This is correct from the code-POV but will probably be a problem for generating release-notes.

**Comment by kaikreuzer** [comment]
No, the release notes are generated from the PR list, so all should be fine.

**Comment by kaikreuzer** [comment]
All, feel free to merge PRs again - the release build is done and the artifacts are published (incl. the [linux packages](https://ci.openhab.org/job/openhab-linuxpkg-release/17/)). I'll compile the release notes tonight and do a small announcement in the forum.

**Comment by wborn** [comment]
Did anyone try 2.5.1 yet? I tried installing a new instance with the Demo package but it fails:

`Suppressed: shaded.org.eclipse.aether.transfer.ArtifactNotFoundException: Could not find artifact javax.json:javax.json-api:jar:1.0 in openhab (https://dl.bintray.com/openhab/mvn/)`

**Comment by kaikreuzer** [comment]
Just fixing it, see https://community.openhab.org/t/openhab-2-5-x-patch-releases/90248/5.
[This change](https://github.com/openhab/openhab-distro/commit/d58c630f189c76862f5f4664caacbc585887bd78) was necessary...

**Comment by kaikreuzer** [comment]
Switching to 2.5.2-SNAPSHOT makes the itests fail, see https://ci.openhab.org/view/Integration%20Builds%20(2.5.x)/job/openHAB2.5.x-Addons/21/console.
Is there anything we can do that this does not happen?

**Comment by Hilbrand** [comment]
Fixed itests: https://github.com/openhab/openhab-addons/pull/6828

**Comment by cweitkamp** [comment]
>  Is there anything we can do that this does not happen?

> Fixed itests: openhab/openhab-addons#6828

I hope we can integrate those steps into the release script. Therefore I created https://github.com/orgs/openhab/projects/3#card-30705443 and asked https://github.com/openhab/openhab-core/pull/1277#issuecomment-565825767.
